labman/db/process.py

```diff
@@ -3230,231 +3235,183 @@ def generate_prep_information(self):
-                    -- mastermix_lot
-                    master_mix_id,
-                    -- water_lot
-                    water_lot_id,
-                    -- processing_robot
-                    gep.epmotion_robot_id AS gepmotion_robot_id,
-                    -- TM300_8_tool
-                    epmotion_tm300_8_tool_id,
-                    -- TM50_8_tool
-                    epmotion_tm50_8_tool_id,
-                    -- sample_plate
-                    p1.external_id AS sample_plate,
-                    -- project_name
-                    study_alias AS project_name,
-                    -- orig_name
-                    sample_id AS orig_name,
-                    -- experiment_design_description
-                    study_description AS experiment_design_description,
-                    -- run_center
-                    'UCSDMI' AS run_center,
-                    -- primer_date
-                    run_date AS primer_date,
-                    -- run_date
-                    '' AS run_date,
-                    -- RUN_PREFIX
-                    '' AS run_prefix,
-                    -- sequencing_meth
-                    'Sequencing by synthesis' AS sequencing_meth,
-                    -- center_name
-                    'UCSDMI' AS center_name,
-                    -- center_project_name
-                    '' AS center_project_name,
-                    -- instrument_model
-                    et.description AS instrument_model,
-                    -- runid
-                    '' AS runid
+
+        # currently, prep files can be created only for amplicon assays
+        if not self.is_amplicon_assay:
+            raise ValueError("Prep file generation is not implemented for {} "
+                             "assays.".format(assay))
+
+        extra_fields = [
+            # 'e'/'r': equipment/reagent
+            ('e', 'lepmotion_robot_id', 'epmotion_robot'),
+            ('e', 'epmotion_tm300_8_tool_id', 'epmotion_tm300_8_tool'),
+            ('e', 'epmotion_tm50_8_tool_id', 'epmotion_tm50_8_tool'),
+            ('e', 'gepmotion_robot_id', 'gdata_robot'),
+            ('e', 'epmotion_tool_id', 'epmotion_tool'),
+            ('e', 'kingfisher_robot_id', 'kingfisher_robot'),
+            ('r', 'extraction_kit_id', 'extraction_kit'),
+            ('r', 'master_mix_id', 'master_mix'),
+            ('r', 'water_lot_id', 'water_lot'),
+        ]
+
+        sql = """
+            -- Naming convention: xcpcp means 'the generic composition
+            -- that is associated to specialized composition aliased as xcp'.
+            -- Likewise, xprpr means 'the generic process that is
+            -- associated with the specialized process aliased as xpr'.
+
+            -- Get the prep sheet info for all wells on any of the library prep
+            -- plates (INCLUDING those that weren't pooled in this pool).
+            SELECT
+                study.study_id, study_sample.sample_id,
```

Comment:
Happily, it is intentional: orig_name defaults to sample_id.  It can end up being set to something different automatically later in the code (if the sample in question is a control) or it can end up being set to something different manually after the prep sheet is generate (if the wet lab/Gail determine that a sample had been mislabeled) but by default is starts out as the same as sample_id.

Reply:
I see, thanks for the explanation.